### PR TITLE
[DeepEP] Implement shared_experts overlap with deepep.combine()

### PR DESCRIPTION
**Pull request by vivekgoe**
### Summary
This PR implements overlap of MOE shared_expert computation with deepep combine communication (during forward pass of training step).  Relates to https://github.com/pytorch/torchtitan/issues/2298 

### Tests
Confirmed expected behavior by taking a profiler trace for reduced DeepSeek-V3-671B model. Also verified loss convergence for 100 training steps.
Screenshot 2026-02-02 at 2 50 58 PM

### CODE REVIEW

**Review by tianyu-l — APPROVED** [review]
sgtm

**Comment by vivekgoe** [comment]
@tianyu-l CI failures seem to be unrelated to PR changes, should I rebase to latest and push another patch?
